File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/Tabs/MatchHistoryTab/Cards/MatchHistoryCardViewContract.java
package com.teamunemployment.lolanalytics.front_page.Tabs.MatchHistoryTab.Cards;

/**
 * @author Josiah Kendall
 *
 * Contract for the cards in the
 */

public interface MatchHistoryCardViewContract {
    void setGraph1(HeadToHeadStat HeadToHeadStat);
    void setGraph2(HeadToHeadStat HeadToHeadStat);
    void setGraph3(HeadToHeadStat HeadToHeadStat);
    void setGraph4(HeadToHeadStat HeadToHeadStat);
    void setHeroChampIcon(String champIconUrl);
    void setEnemyChampIcon(String champIconUrl);
    void setTimeStamp(String timeStamp);
    void setResult(boolean win);
    void setChamp(String champName);
    void setSummaryChart(HeadToHeadStat HeadToHeadStat);
}

File: app/src/main/java/com/teamunemployment/lolanalytics/front_page/tabs/MatchHistoryTab/Cards/MatchHistoryCardViewContract.kt
package com.teamunemployment.lolanalytics.front_page.tabs.MatchHistoryTab.Cards

/**
 * @author Josiah Kendall
 *
 * Contract for the cards in the
 */

interface MatchHistoryCardViewContract {

    fun setDetailsUrl(detailsUrl : String)
    fun getDetailsUrl() : String
    fun setGraph1(HeadToHeadStat: HeadToHeadStat)
    fun setGraph2(HeadToHeadStat: HeadToHeadStat)
    fun setGraph3(HeadToHeadStat: HeadToHeadStat)
    fun setGraph4(HeadToHeadStat: HeadToHeadStat)
    fun setHeroChampIcon(champIconUrl: String)
    fun setEnemyChampIcon(champIconUrl: String)
    fun setTimeStamp(timeStamp: String)
    fun setResult(result: String)
    fun setChamp(champName: String)
    fun setSummaryChart(HeadToHeadStat: HeadToHeadStat)
    fun setChampPlaceHolders()
}
